src/Psalm/Internal/Type/Comparator/AtomicTypeComparator.php

```diff
@@ -191,7 +192,12 @@ public static function isContainedBy(
         }
 
         if (($container_type_part instanceof TCallable
-                && $input_type_part instanceof TCallable)
+            && ($input_type_part instanceof TCallable
+                || $input_type_part instanceof TCallableArray
+                || $input_type_part instanceof TCallableObject
+                || $input_type_part instanceof TCallableString
+                || $input_type_part instanceof TCallableKeyedArray
```

Comment:
Looks like `TCallableInterface` would help to prevent such cases. And would help to make clearer distinction between `callable` and it's implementations.

Reply:
+1